shared/test/test_sources.rb

```diff
@@ -471,6 +471,137 @@ def test_restore_main_json_with_null_animation_versions
     assert_restores_main_json_with_animation_version nil
   end
 
+  def test_remix_source_file
+    # Mock destination
+    @destination_channel = create_channel
+    @destination_api = FilesApiTestHelper.new(current_session, 'sources', @destination_channel)
+    @destination_animation_api = FilesApiTestHelper.new(current_session, 'animations', @destination_channel)
+
+    # Create two animations
+    animation_key_1 = @api.add_random_suffix('animation-key')
+    animation_key_2 = @api.add_random_suffix('animation-key-1')
+    animation_filename_1 = "#{animation_key_1}.png"
+    animation_filename_2 = "#{animation_key_2}.png"
+    delete_all_animation_versions(animation_filename_1)
+    delete_all_animation_versions(animation_filename_2)
+
+    # Upload the two animations
+    animation_1 = 'stub-png-1'
+    animation_2 = 'stub-png-2'
+    animation_1_vid = put_animation(animation_filename_1, animation_1)
+    animation_2_vid = put_animation(animation_filename_2, animation_2)
+
+    # Update main.json
+    main_json_v1 = {
+      "source": "//version 1",
+      "animations": {
+        "orderedKeys": [animation_key_1, animation_key_2],
+        "propsByKey": {
+          "#{animation_key_1}": {
+            "name": "Remix First",
+            "sourceUrl": nil,
+            "version": animation_1_vid
+          },
+          "#{animation_key_2}": {
+            "name": "Remix Second",
+            "sourceUrl": nil,
+            "version": animation_2_vid
```

Comment:
Possible addition to this test case: Add a library animation, make sure it shows up in the final manifest with the same sourceUrl / version.